cmd/main.go

```diff
@@ -744,158 +452,448 @@ func Run() error {
+	logTextileOnly := logCmd.Flag("textile-only", "Whether to list/change only Textile subsystems, or all available subsystems").Short('t').Bool()
+	cmds[logCmd.FullCommand()] = func () error {
 		return Logs(*logSubsystem, *logLevel, *logTextileOnly)
+	}
+
+	// ================================
 
 	// message
-	case messageAddCmd.FullCommand():
+	messageCmd := appCmd.Command("message", "Manage Textile Messages").Alias("messages")
+
+	// message add
+	messageAddCmd      := messageCmd.Command("add", "Adds a message to a thread")
+	messageAddThreadID := messageAddCmd.Arg("thread", "Thread ID").Required().String()
+	messageAddBody     := messageAddCmd.Arg("body", "The message to add the thread").Required().String()
+	cmds[messageAddCmd.FullCommand()] = func () error {
 		return MessageAdd(*messageAddThreadID, *messageAddBody)
+	}
 
-	case messageListCmd.FullCommand():
+	// message list
+	messageListCmd      := messageCmd.Command("list", "Paginates thread messages").Alias("ls").Default()
+	messageListThreadID := messageListCmd.Arg("thread", "Thread ID, omit to paginate all messages").String()
+	messageListOffset   := messageListCmd.Flag("offset", "Offset ID to start the listing from").Short('o').String()
+	messageListLimit    := messageListCmd.Flag("limit", "List page size").Default("10").Short('l').Int()
+	cmds[messageListCmd.FullCommand()] = func () error {
 		return MessageList(*messageListThreadID, *messageListOffset, *messageListLimit)
+	}
 
-	case messageGetCmd.FullCommand():
+	// message get
+	messageGetCmd     := messageCmd.Command("get", "Gets a message by its own Block ID")
+	messageGetBlockID := messageGetCmd.Arg("message-block", "Message Block ID").String()
+	cmds[messageGetCmd.FullCommand()] = func () error {
 		return MessageGet(*messageGetBlockID)
+	}
 
-	case messageIgnoreCmd.FullCommand():
+	// message ignore
+	messageIgnoreCmd     := messageCmd.Command("ignore", "Ignores a message by its own Block ID").Alias("remove").Alias("rm")
+	messageIgnoreBlockID := messageIgnoreCmd.Arg("message-block", "Message Block ID").String()
+	cmds[messageIgnoreCmd.FullCommand()] = func () error {
 		return MessageIgnore(*messageIgnoreBlockID)
+	}
+
+	// ================================
 
 	// migrate
-	case migrateCmd.FullCommand():
-		repoPath, err := getRepoPath(*migrateRepoPath)
+	migrateCmd  := appCmd.Command("migrate", "Migrate the node repository and exit")
+	migrateRepo := migrateCmd.Flag("repo", "Specify a custom repository path").Short('r').String()
+	migratePin  := migrateCmd.Flag("pin", "Specify the pin for datastore encryption, omit if no pin was used during init").Short('p').String()
+	cmds[migrateCmd.FullCommand()] = func () error {
+		repo, err := getRepo(*migrateRepo)
 		if err != nil {
 			return err
 		}
-		return Migrate(repoPath, *migratePinCode)
+		return Migrate(repo, *migratePin)
+	}
+
+	// ================================
+	// @todo are notifications blocks?
```

Comment:
Nope, they are local-only and are generated from most block updates, e.g., https://github.com/textileio/go-textile/blob/master/core/threads_service.go#L395.